pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -209,11 +209,29 @@ type EC2Metadata interface {
 	GetMetadata(path string) (string, error)
 }
 
+// AWS volume types
+const (
+	// Provisioned IOPS SSD
+	VolumeTypeIO1 = "io1"
+	// General Purpose SSD
+	VolumeTypeGP2 = "gp2"
+	// Cold HDD (sc1)
+	VolumeTypeSC1 = "sc1"
+	// Throughput Optimized HDD
+	VolumeTypeST1 = "st1"
+)
+
 // VolumeOptions specifies capacity and tags for a volume.
 type VolumeOptions struct {
-	CapacityGB int
-	Tags       map[string]string
-	PVCName    string
+	CapacityGB       int
+	Tags             map[string]string
+	PVCName          string
+	VolumeType       string
+	AvailabilityZone string
+	// IOPSPerGB x CapacityGB will give total IOPS of the volume to create.
+	// IOPSPerGB must be bigger than zero and smaller or equal to 30.
+	// Calculated total IOPS will be capped at 20000 IOPS.
+	IOPSPerGB int
```

Comment:
@justinsb If iops was fixed in the storage class it would not scale with PVC sizes. If say 1000 IOPS was configured, it would be really slow volume for someone who wants 500 GB disk and quite a fast one for  1GB disk.

With IOPSPerGB, admin can create a "fast" class, which is fast for everybody, regardless of PVC size.